electron_app/UploadManager.js

```diff
@@ -3,59 +3,66 @@ const mime = require('mime-types');
-            file.progress = Math.round(progress.percentage);
-            file.transferred = progress.transferred;
-            win.webContents.send(constants.IPC_UPLOAD_STATUS, file);
-        });
         req.post({
             headers: {
-                    'Authorization': 'Bearer ' + access_token,
-                    'X-Upload-Content-Type': contentType,
-                    'X-Upload-Content-Length': size
-                },
-                uri: url + 'Uploads/' + file.name,
-                method: 'POST',
-                json: false,
-                resolveWithFullResponse: true
+                'Authorization': 'Bearer ' + access_token,
+                'X-Upload-Content-Type': contentType,
+                'X-Upload-Content-Length': size
             },
+            uri: url + 'Uploads/' + file.name,
+            method: 'POST',
+            json: false,
+            resolveWithFullResponse: true
+        },
             function (err, resp, body) {
                 if (err) {
                     console.log('Error!', err);
                 } else {
                     var location = resp.headers['location'];
-                    if(location !== undefined) {
-                        var reqConfig = req.put({uri: location},
+                    if (location !== undefined) {
+                        var reqConfig = req.put({ uri: location },
                             function (err, resp, body) {
-                              if (err) {
-                                  console.log('Error!', err);
-                              } else {
-                                  console.log('URL: ' + body);
-                              }
-                          });
-                          reqConfig.setHeader('Content-Length', size);
-                          fs.createReadStream(file.path).pipe(progressConf).pipe(reqConfig);
+                                if (err) {
+                                    console.log('Error!', err);
+                                } 
+                            });
+                        reqConfig.setHeader('Content-Length', size);
+                        fs.createReadStream(file.path).pipe(progressConf).pipe(reqConfig);
+                        requestList.push(reqConfig);
+
                     } else {
                         console.log(JSON.stringify(resp));
                         console.error("There was an error trying to connect to google");
                     }
-                    
+
                 }
+                progressConf.on('progress', function (progress) {
+                    file.progress = Math.round(progress.percentage);
+                    file.transferred = progress.transferred;
+                    win.webContents.send(constants.IPC_UPLOAD_STATUS, file);
+                });
             }
         );
     })
 }
 
-module.exports = uploadManager;
+const uploadManagerCancel = () => {
+    if (requestList !== null) {
```

Comment:
It looks like `requestList` will never be null, since it's initialized, is this check necessary?